src/log.rs

```diff
@@ -574,6 +713,11 @@ impl Log {
 		}
 
 		let reading = self.reading.write();
+		if reading.is_none() {
```

Comment:
I am wondering if it would be possible to drop `ReadingState::Reading` in favor to only check if reading is_some.
In fact, I am also wondering if a `pending_reading` vecdeque of files (I am not really thinking size of log could be an issue).
That way threads will generally all wait on their own list of content, making things simplier than flush_one file rotation.
But could also lead to things I did not foresee (l m thinking appending flush delay).

Reply:
This will require more refactoring. Currently when the log is read to the end `reading` is not immediately reset to `None`. This happens later in the flushing thread.  As for the `pending_reading` queue, there's always only one pending unread log by design. Size of unread log is indeed an issue, as all unread changes are also kept in memory, in the overlay. We can't just keep writing to log files with reades falling behind, as this will require a lot of memory.